devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java

```diff
@@ -162,16 +165,26 @@ public void execute() throws MojoExecutionException {
         }
         File projectRoot = outputDirectory;
         File pom = new File(projectRoot, "pom.xml");
+        Model parentPomModel = null;
 
         if (pom.isFile()) {
-            throw new MojoExecutionException("Unable to generate the project in a directory that already contains a pom.xml");
+            try {
+                parentPomModel = MojoUtils.readPom(pom);
+                if (!"pom".equals(parentPomModel.getPackaging())) {
+                    throw new MojoExecutionException("The parent project must have a packaging type of POM. Current packaging: "
+                            + parentPomModel.getPackaging());
+                }
+                askTheUserForMissingValues();
+            } catch (IOException e) {
+                throw new MojoExecutionException("Could not access parent pom.", e);
+            }
         } else {
             askTheUserForMissingValues();
-            projectRoot = new File(outputDirectory, projectArtifactId);
-            if (projectRoot.exists()) {
-                throw new MojoExecutionException("Unable to create the project, " +
-                        "the directory " + projectRoot.getAbsolutePath() + " already exists");
-            }
```

Comment:
Why are we changing the behavior for non pom project?

Reply:
Actually the else block is useless and can be re-written as:
```
if (pom.isFile()) {
    try {
        parentPomModel = MojoUtils.readPom(pom);
        if (!"pom".equals(parentPomModel.getPackaging())) {
            throw new MojoExecutionException("The parent project must have a packaging type of POM. Current packaging: "
                    + parentPomModel.getPackaging());
        }
    } catch (IOException e) {
        throw new MojoExecutionException("Could not access parent pom.", e);
    }
}
askTheUserForMissingValues();
projectRoot = new File(outputDirectory, projectArtifactId);
if (projectRoot.exists()) {
    throw new MojoExecutionException("Unable to create the project, " +
            "the directory " + projectRoot.getAbsolutePath() + " already exists");
}
```

Reply:
Right. +1

Reply:
@ghokun well I think we should still block creation on existing gradle project no?

Reply:
In my previous comment, I mentioned there are different cases to be considered:
https://github.com/quarkusio/quarkus/pull/12916#issuecomment-716412935

There is also another case: Parent project could containt both maven and gradle files.

https://github.com/quarkusio/quarkus/issues/12829#issuecomment-716137809
Based on this comment, I continued with maven-parent, maven-submodule combination.

We can extend the scope of this PR to include all cases.

Reply:
@ghokun that's what I mean't, we should keep failing when we don't have a pom until a possible PR to deal with those case..

Reply:
@ia3andy
This is the matrix I came up with:
```
Build Tool MAVEN:
no file                                 : Generate
ONLY pom.xml                            : Check packaging of pom.xml -> if packaging is pom, add module and generate else fail
ONLY gradle files                       : Fail with error message (You are trying to create a maven project inside a gradle project etc..)
pom.xml and other possible gradle files : Check packaging of pom.xml -> if packaging is pom, add module and generate else fail

Build Tool GRADLE/GRADLE_KOTLIN_DSL:
no file                                 : Generate
ONLY gradle files                       : Fail (Since not implemented)
ONLY pom.xml                            : Fail with error message (You are trying to create a gradle project inside a maven project etc..)
gradle files and pom.xml                : Fail (Since not implemented)
```

Reply:
@ghokun looks good!